src/chttpd/src/chttpd.erl

```diff
@@ -681,8 +681,10 @@ json_body(Httpd) ->
     case body(Httpd) of
         undefined ->
             throw({bad_request, "Missing request body"});
+        Body when is_binary(Body) ->
+            ?JSON_DECODE(maybe_decompress(Httpd, Body));
         Body ->
```

Comment:
Let's explicitly match on `{Props} when is_list(Props)` (or similar ejson match) to assert that the only other option is ejson here.

Reply:
the `{Props}` wouldn't work for the case when JSON term is a list of items.

We have `{Props}` match in `json_body_obj/1` bellow.

Reply:
Ah good point we do match on that in json_body_obj.

In this function, if we support arbitrary ejson values there would still be an ambiguity between a terminal string value say `<<"a">>` and the encoded form `<<"\"a\"">>` as they'd both be binaries. We have that case in the `_config` handler to set values. I think that just means in that case if we want to override req_body, we'd have to supply the encoded form.

Reply:
Good point. I think the easiest way to support this corner case would be to bypass call to body.
```

body(#httpd{req_body=undefined}) ->
    case body(Httpd) of
         ...
     end;
body(#httpd{req_body=ReqBody}) ->
    ReqBody.
   
```

Reply:
That's better solution and removes the ugly hack with post_body